### PR TITLE
[stable/mongodb-replicaset] Update MongoDB Exporter to 0.10.x

### PR DESCRIPTION
**Pull request by ghost**
#### What this PR does / why we need it:
It allows the chart to use the MongoDB exporter in the latest version.

#### Special notes for your reviewer:
In the exporter version 0.10.0 the support for TLS related cli options is dropped in favor of using the connection string. To use the new exporter version I had to introduce a little helper function that generates the MongoDB connection string. 

I also dropped the fsGroup setting, since it threw unsupported errors and I agree that this setting at this level doesn't do anything.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml [hunk 1]
```diff
@@ -364,6 +345,6 @@ spec:
       {{- end }}
       {{- end }}
 {{- else }}
-        - name: datadir
-          emptyDir: {}
+      - name: datadir
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: This should probably be undone as now datadir doesn't line up with configdir above

File: stable/mongodb-replicaset/templates/_helpers.tpl [hunk 2]
```diff
@@ -61,3 +61,18 @@ Create the name for the key secret.
         {{- template "mongodb-replicaset.fullname" . -}}-keyfile
     {{- end -}}
 {{- end -}}
+
+{{- define "mongodb-replicaset.connection-string"  -}}
+  {{- $string := "" -}}
+  {{- if .Values.auth.enabled }}
+   {{- $string = printf "mongodb://$METRICS_USER:$METRICS_PASSWORD@localhost:%s" (.Values.port|cat) -}}
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/templates/_helpers.tpl]: I'm not familiar with this cat command. Do you have any documentation on it? What does it do here that can't be accomplished with `printf "....%d" .Values.port -}}`?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml (hunk 1 above)
Comment (ghost) [reply stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: You are right, my IDE mislead me here.

File: stable/mongodb-replicaset/templates/_helpers.tpl (hunk 2 above)
Comment (ghost) [reply stable/mongodb-replicaset/templates/_helpers.tpl]: I had troube with using printf %d. It always lead to very strange output. cat in this case makes sure that the output is always interpreted as a string. Which worked for me in this situation.

`--mongodb.uri mongodb://$METRICS_USER:$METRICS_PASSWORD@localhost:%!!(MISSING)d(float64=27017)`

Comment (ghost) [reply stable/mongodb-replicaset/templates/_helpers.tpl]: Switched it to `toString` which is  a lot more self explanatory at least.

**Review by steven-sheehy — COMMENTED** [review]
/assign
/ok-to-test

**Comment by ofcourseican** [comment]
`Error: Could not get apiVersions from Kubernetes: unable to retrieve the complete list of server APIs: metrics.k8s.io/v1beta1: the server is currently unable to handle the request`

@steven-sheehy Looks like there is something wrong with the testing infrastructure. Not sure what I'm supposed to do now.

**Review by steven-sheehy — APPROVED** [review]
/lgtm
Thanks!

**Pull request merged**